test-specifications/token-service/tokenCreateTransaction.md

```diff
@@ -0,0 +1,1150 @@
+
+#### JSON Request Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 641,
+  "method": "createToken",
+  "params": {
+    "name": "testname",
+    "symbol": "testsymbol",
+    "treasuryAccount": "0.0.2"
+  }
+}
+```
+
+#### JSON Response Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 641,
+  "result": {
+    "tokenId": "0.0.836",
+    "status": "SUCCESS"
+  }
+}
+```
+
+### **Decimals:**
+
+- The number of decimal places by which a token is divisible.
+
+| Test no | Name                                                                           | Input                                                                                                    | Expected response                                                                       | Implemented (Y/N) |
+|---------|--------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------|-------------------|
+| 1       | Creates a fungible token with a valid amount of decimals                       | name="testname", symbol="testsymbol", decimals=3, treasuryAccount=<OPERATOR_ACCOUNT_ID>                  | The token creation succeeds and the token has 3 decimals.                               | N                 |
+| 2       | Creates a fungible token with the minimum amount of decimals                   | name="testname", symbol="testsymbol", decimals=0, treasuryAccount=<OPERATOR_ACCOUNT_ID>                  | The token creation succeeds and the token has 0 decimals.                               | N                 |
+| 3       | Creates a fungible token with a decimal amount below the minimum amount        | name="testname", symbol="testsymbol", decimals=-1, treasuryAccount=<OPERATOR_ACCOUNT_ID>                 | The token creation fails with an INVALID_TOKEN_DECIMALS response code from the network. | N                 |
+| 4       | Creates a fungible token with the maximum amount of decimals                   | name="testname", symbol="testsymbol", decimals=2147483647, treasuryAccount=<OPERATOR_ACCOUNT_ID>         | The token creation succeeds and the token has 2,147,483,647 decimals.                   | N                 |
+| 5       | Creates a fungible token with a decimal amount that exceeds the maximum amount | name="testname", symbol="testsymbol", decimals=2147483648, treasuryAccount=<OPERATOR_ACCOUNT_ID>         | The token creation fails with an INVALID_TOKEN_DECIMALS response code from the network. | N                 |
```

Comment:
It _might_ be worth passing a value equal to 2<sup>32</sup>-1 as well, to ensure the consensus node handles the edges.

Reply:
Isn't this and test no. 3 already testing the edges?

Reply:
Some, yes, but not all.  In particular this value is the _signed_ int max; but we should _also_ test _unsigned_ int max (for unsigned values) or `-1` (for signed values).

Reply:
So since this is an unsigned value, won't the `-1` fit the bill for testing the unsigned int max? It's interesting here because this value is a `uint32` in the protobuf, however it appears to be read as an `int32` on the services side.

Reply:
Java doesn't support unsigned integers, but protobuf does.
Protobuf works around this by copying the value into a Java signed integer bit-for-bit and leaving it up to the Java code to handle overflow and underflow.
This approach, thanks to twos-compliment encoding, results in -1 in Java being bitwise identical to UNSIGNED_INT_MAX.  The Java code has to handle overflow correctly, so the TCK should, generally, test, at minimum, all permissible (in protobuf) values from the set `[0, -1, signed_int_max +/- 1, unsigned_int_max, unsigned_int_max - 1, signed_int_min, and signed_int_min + 1]`.

Reply:
Added more test cases for int types, let me know what you think